### PR TITLE
Port monotouch-test suite to Xamarin.Mac

### PR DESCRIPTION
**Pull request by timrisi**

### CODE REVIEW

File: tests/monotouch-test/Security/ImportExportTest.cs [hunk 1]
```diff
@@ -79,8 +88,12 @@ public void P12_AuthFailed ()
 		{
 			using (NSMutableDictionary options = new NSMutableDictionary ()) {
 				options [SecImportExport.Passphrase] = new NSString ("b5");
-				NSDictionary[] array;
+				NSDictionary [] array;
```
Comment (spouliot) [inline tests/monotouch-test/Security/ImportExportTest.cs]: unneeded change

File: tests/monotouch-test/Security/ImportExportTest.cs [hunk 2]
```diff
@@ -65,8 +65,17 @@ public void P12_Success ()
 				case "chain":
 					Assert.True (x.Value is NSArray, "NSArray");
 					NSArray a = (x.Value as NSArray);
-					Assert.That (a.Count, Is.EqualTo ((nuint) 1), "Count");
+					Assert.That (a.Count, Is.EqualTo ((nuint)1), "Count");
```
Comment (spouliot) [inline tests/monotouch-test/Security/ImportExportTest.cs]: unneeded change

File: tests/monotouch-test/SpriteKit/SK3DNodeTest.cs [hunk 3]
```diff
@@ -43,19 +45,25 @@ public void VersionCheck ()
...
+			Assert.Ignore ("This doesn't seem to work properly in macOS 10.12");
+#else
 			if (UIDevice.CurrentDevice.CheckSystemVersion (9, 0))
 				Assert.Ignore ("This doesn't seem to work properly in the iOS 9");
-			
+#endif
+
 			// SK3Node loads SCNRenderer dynamically, so make sure it's actually loaded.
-			GC.KeepAlive (Class.GetHandle (typeof(SCNRenderer)));
+			GC.KeepAlive (Class.GetHandle (typeof (SCNRenderer)));
```
Comment (spouliot) [inline tests/monotouch-test/SpriteKit/SK3DNodeTest.cs]: unneeded change

File: tests/monotouch-test/System.Net.Http/MessageHandlers.cs [hunk 4]
```diff
@@ -29,7 +29,9 @@ HttpMessageHandler GetHandler (Type handler_type)
 		[TestCase (typeof (HttpClientHandler))]
 		[TestCase (typeof (CFNetworkHandler))]
 #endif
+#if !MONOMAC
```
Comment (spouliot) [inline tests/monotouch-test/System.Net.Http/MessageHandlers.cs]: why ? NSUrlSessionHandler exists on XM (but it's in a different assembly IIRC)

File: tests/monotouch-test/VideoToolbox/VTCompressionSessionTests.cs [hunk 5]
```diff
@@ -127,6 +126,9 @@ public void CompressionSessionGetSupportedPropertiesTest ()
 		// This test is (kind of) expected to be null due to as of iOS 8 all supported properties are not meant to be serialized
 		// see CompressionSessionGetSupportedPropertiesTest.
 		[Test]
+#if MONOMAC
+		[Ignore ("Crashes with SIGSEGV when trying to dispose session after calling session.GetSerializableProperties ()")]
```
Comment (spouliot) [inline tests/monotouch-test/VideoToolbox/VTCompressionSessionTests.cs]: please file a bug report ^ so we do not forget about it. It might be something that affect iOS too (and we just lucky to avoid the crash)

File: tests/monotouch-test/CoreFoundation/BundleTest.cs [hunk 6]
```diff
@@ -86,14 +90,18 @@ public void TestGetMain ()
 		public void TestBuiltInPlugInsUrl ()
 		{
 			var main = CFBundle.GetMain ();
-			Assert.That(main.BuiltInPlugInsUrl.ToString (), Contains.Substring ("PlugIns/"));
+			Assert.That (main.BuiltInPlugInsUrl.ToString (), Contains.Substring ("PlugIns/"));
```
Comment (spouliot) [inline tests/monotouch-test/CoreFoundation/BundleTest.cs]: unneeded change

File: tests/monotouch-test/CoreFoundation/BundleTest.cs [hunk 7]
```diff
@@ -13,14 +12,17 @@
...
-namespace MonoTouchFixtures.CoreFoundation {
+namespace MonoTouchFixtures.CoreFoundation
+{
 
 	[TestFixture]
 	[Preserve (AllMembers = true)]
-	public class BundleTest {
-
+	public class BundleTest
+	{
```
Comment (spouliot) [inline tests/monotouch-test/CoreFoundation/BundleTest.cs]: that's not the mono convention - and it's also a noisy change

File: tests/monotouch-test/CoreGraphics/CGImageTest.cs [hunk 8]
```diff
@@ -22,21 +26,26 @@
...
+namespace MonoTouchFixtures.CoreGraphics
+{
+
 	[TestFixture]
 	[Preserve (AllMembers = true)]
-	public class CGImageTest {
-		
+	public class CGImageTest
+	{
+
```
Comment (spouliot) [inline tests/monotouch-test/CoreGraphics/CGImageTest.cs]: that's not the mono convention - and it's also a noisy change

File: tests/monotouch-test/CoreGraphics/CGImageTest.cs [hunk 9]
```diff
@@ -22,21 +26,26 @@
 #endif
 using NUnit.Framework;
 
-namespace MonoTouchFixtures.CoreGraphics {
-	
+namespace MonoTouchFixtures.CoreGraphics
+{
```
Comment (spouliot) [inline tests/monotouch-test/CoreGraphics/CGImageTest.cs]: that's not the mono convention - and it's also a noisy change

File: tests/monotouch-test/CoreImage/CIKernelTests.cs [hunk 10]
```diff
@@ -142,24 +147,33 @@ public void CIKernel_BasicTest ()
...
+
 
 					foreach (CustomerFilterType type in Enum.GetValues(typeof(CustomerFilterType))) {
 						MyCustomFilter filter = new MyCustomFilter (type);
 						filter.MyImage = ciImg;
 
 						CIImage outputImage = filter.OutputImage;
 
-						CGImage cgImage = context.CreateCGImage (outputImage, outputImage.Extent);
+					CGImage cgImage = context.CreateCGImage (outputImage, outputImage.Extent);
```
Comment (spouliot) [inline tests/monotouch-test/CoreImage/CIKernelTests.cs]: indent

File: tests/monotouch-test/Foundation/NetServiceTest.cs [hunk 11]
```diff
@@ -28,7 +32,14 @@ public class NetServiceTest {
 		[Test]
 		public void DefaultCtor ()
 		{
-			using (var ns = new NSNetService ("d", "_test._tcp", UIDevice.CurrentDevice.Name, 1234)) {
+			using (var ns = new NSNetService ("d", 
+			                                  "_test._tcp",
+#if MONOMAC
+			                                  "DeviceName",
+#else
+											  UIDevice.CurrentDevice.Name,
```
Comment (spouliot) [inline tests/monotouch-test/Foundation/NetServiceTest.cs]: indent

File: tests/monotouch-test/AudioToolbox/FourCCTest.cs [hunk 12]
```diff
@@ -74,12 +74,15 @@ public void AudioQueuePropertyTest ()
 		[Test]
 		public void AudioQueueTimePitchAlgorithmtest ()
 		{
-			Assert.That ((int) AudioQueueTimePitchAlgorithm.Spectral, Is.EqualTo (FourCC ("spec")), "Spectral");
-			Assert.That ((int) AudioQueueTimePitchAlgorithm.TimeDomain, Is.EqualTo (FourCC ("tido")), "TimeDomain");
+			Assert.That ((int)AudioQueueTimePitchAlgorithm.Spectral, Is.EqualTo (FourCC ("spec")), "Spectral");
```
Comment (dalexsoto) [inline tests/monotouch-test/AudioToolbox/FourCCTest.cs]: not needed whitespace change

File: tests/monotouch-test/AudioToolbox/FourCCTest.cs [hunk 13]
```diff
@@ -74,12 +74,15 @@ public void AudioQueuePropertyTest ()
...
+			Assert.That ((int)AudioQueueTimePitchAlgorithm.TimeDomain, Is.EqualTo (FourCC ("tido")), "TimeDomain");
```
Comment (dalexsoto) [inline tests/monotouch-test/AudioToolbox/FourCCTest.cs]: not needed whitespace change

File: tests/common/mac/MacTestMain.cs [hunk 14]
```diff
@@ -53,3 +53,18 @@ public void Shutdown()
 		}
 	}
 }
+
+partial class TestRuntime
+{
```
Comment (dalexsoto) [inline tests/common/mac/MacTestMain.cs]: Not following mono conventions

File: tests/monotouch-test/CoreFoundation/BundleTest.cs [hunk 15]
```diff
@@ -13,14 +12,17 @@
 #endif
 using NUnit.Framework;
 
-namespace MonoTouchFixtures.CoreFoundation {
+namespace MonoTouchFixtures.CoreFoundation
+{
```
Comment (dalexsoto) [inline tests/monotouch-test/CoreFoundation/BundleTest.cs]: Not following mono conventions

File: tests/monotouch-test/CoreImage/CoreContextTest.cs [hunk 16]
```diff
@@ -66,14 +75,17 @@ public void CreateRefCount ()
...
+#if !MONOMAC // CreateCGImage returning null on mac
+				using (var v = ctx.CreateCGImage (img, new RectangleF (0, 0, 5, 5), CIFormat.ARGB8, null, false)) {
 					int rc = Messaging.int_objc_msgSend (v.Handle, retainCount.Handle);
 					Assert.AreEqual (1, rc, "CreateCGImage #c1");
 				}
+#endif
 			}
 		}
 
+#if !MONOMAC // No EAGLContext for Mac
```
Comment (dalexsoto) [inline tests/monotouch-test/CoreImage/CoreContextTest.cs]: Really? wasn't this inside the OpenTK dll? (I could be wrong)

File: tests/monotouch-test/CoreMedia/CMTimeTests.cs [hunk 17]
```diff
@@ -127,19 +133,25 @@ public void CMTimeRangeConstants ()
 #if !XAMCORE_3_0
 			Assert.DoesNotThrow (() => { var x = CMTimeRange.Invalid; }, "CMTimeRangeConstants - Invalid");
 #endif
+#if !MONOMAC
 			if (UIDevice.CurrentDevice.CheckSystemVersion (9, 0)) {
-				Assert.DoesNotThrow (() => { var x = CMTimeRange.InvalidMapping; }, "CMTimeRangeConstants - InvalidMapping");
-				Assert.DoesNotThrow (() => { var x = CMTimeRange.TimeMappingSourceKey; }, "CMTimeRangeConstants - TimeMappingSourceKey");
-				Assert.DoesNotThrow (() => { var x = CMTimeRange.TimeMappingTargetKey; }, "CMTimeRangeConstants - TimeMappingTargetKey");
+#endif
+			Assert.DoesNotThrow (() => { var x = CMTimeRange.InvalidMapping; }, "CMTimeRangeConstants - InvalidMapping");
```
Comment (dalexsoto) [inline tests/monotouch-test/CoreMedia/CMTimeTests.cs]: indentation

File: tests/monotouch-test/Foundation/UbiquitousKeyValueStoreTest.cs [hunk 18]
```diff
@@ -32,8 +32,6 @@ public class NSUbiquitousKeyValueStoreTest {
 		[Test]
 		public void Indexer ()
 		{
-			Assert.Ignore ("Doesn't work on watchOS");
```
Comment (dalexsoto) [inline tests/monotouch-test/Foundation/UbiquitousKeyValueStoreTest.cs]: Why this got removed?

File: tests/monotouch-test/GameController/GamepadSnapshotTest.cs [hunk 19]
```diff
@@ -21,17 +23,21 @@
 #endif
 using NUnit.Framework;
 
-namespace MonoTouchFixtures.GameController {
+namespace MonoTouchFixtures.GameController
+{
```
Comment (dalexsoto) [inline tests/monotouch-test/GameController/GamepadSnapshotTest.cs]: unnecessary change, not following mono conventions

File: tests/monotouch-test/GameController/GamepadSnapshotTest.cs [hunk 20]
```diff
@@ -21,17 +23,21 @@
...
 
 	[TestFixture]
 	[Preserve (AllMembers = true)]
-	public class GamepadSnapshotTest {
+	public class GamepadSnapshotTest
+	{
```
Comment (dalexsoto) [inline tests/monotouch-test/GameController/GamepadSnapshotTest.cs]: unnecessary change, not following mono conventions

File: tests/monotouch-test/ImageIO/CGImageSourceTest.cs [hunk 21]
```diff
@@ -18,73 +17,109 @@
 #endif
 using NUnit.Framework;
 
-namespace MonoTouchFixtures.ImageIO {
-	
+namespace MonoTouchFixtures.ImageIO
+{
```
Comment (dalexsoto) [inline tests/monotouch-test/ImageIO/CGImageSourceTest.cs]: not needed change

File: tests/monotouch-test/ImageIO/CGImageSourceTest.cs [hunk 22]
```diff
@@ -18,73 +17,109 @@
...
+
 	[TestFixture]
 	[Preserve (AllMembers = true)]
-	public class CGImageSourceTest {
+	public class CGImageSourceTest
```
Comment (dalexsoto) [inline tests/monotouch-test/ImageIO/CGImageSourceTest.cs]: not needed change

File: tests/monotouch-test/ModelIO/MDLLight.cs [hunk 23]
```diff
@@ -32,15 +34,21 @@
 using OpenTK;
 using NUnit.Framework;
 
-namespace MonoTouchFixtures.ModelIO {
+namespace MonoTouchFixtures.ModelIO
+{
```
Comment (dalexsoto) [inline tests/monotouch-test/ModelIO/MDLLight.cs]: not needed change

File: tests/monotouch-test/ModelIO/MDLLight.cs [hunk 24]
```diff
@@ -32,15 +34,21 @@
...
 
 	[TestFixture]
 	// we want the test to be available if we use the linker
 	[Preserve (AllMembers = true)]
-	public class MDLLightTest {
+	public class MDLLightTest
+	{
```
Comment (dalexsoto) [inline tests/monotouch-test/ModelIO/MDLLight.cs]: not needed change

File: tests/monotouch-test/ObjCRuntime/RegistrarTest.cs [hunk 25]
```diff
@@ -1715,9 +1737,11 @@ public new void IdAsIntPtr (IntPtr id)
 				Assert.AreEqual (IntPtr.Zero, id, "Zero");
 			}
 		}
+#endif
 
-#if !__TVOS__ && !__WATCHOS__
-		class Test24970 : UIApplicationDelegate {
+#if !__TVOS__ && !__WATCHOS__ && !MONOMAC
+		class Test24970 : UIApplicationDelegate
+		{
```
Comment (dalexsoto) [inline tests/monotouch-test/ObjCRuntime/RegistrarTest.cs]: not needed change

File: tests/monotouch-test/Security/KeyTest.cs [hunk 26]
```diff
@@ -366,8 +388,12 @@ public void RSA ()
 		{
 			TestRuntime.AssertXcodeVersion (8, 0);
 			NSError error;
+#if MONOMAC
+			using (var key = SecKey.CreateRandomKey (SecKeyType.RSA, 1024, null, out error)) {
+#else
 			using (var key = SecKey.CreateRandomKey (SecKeyType.RSA, 512, null, out error)) {
-				Assert.Null (error, "RSA/error");
+#endif
+			Assert.Null (error, "RSA/error");
```
Comment (dalexsoto) [inline tests/monotouch-test/Security/KeyTest.cs]: indentation

File: tests/monotouch-test/SystemConfiguration/NetworkReachabilityTest.cs [hunk 27]
```diff
@@ -99,10 +101,12 @@ public void CtorIPAddressPair ()
...
 				// using Loopback iOS 10 / tvOS 10 returns no flags (0) on devices
-				if ((Runtime.Arch == Arch.DEVICE) && TestRuntime.CheckXcodeVersion (8,0))
-					Assert.That ((int) flags, Is.EqualTo (0), "#1 Reachable");
+#if !MONOMAC
+				if ((Runtime.Arch == Arch.DEVICE) && TestRuntime.CheckXcodeVersion (8, 0))
+					Assert.That ((int)flags, Is.EqualTo (0), "#1 Reachable");
 				else
-					Assert.That (flags, Is.EqualTo (NetworkReachabilityFlags.Reachable), "#1 Reachable");
+#endif
+				Assert.That (flags, Is.EqualTo (NetworkReachabilityFlags.Reachable), "#1 Reachable");
```
Comment (dalexsoto) [inline tests/monotouch-test/SystemConfiguration/NetworkReachabilityTest.cs]: indentation

File: tests/monotouch-test/CoreImage/CoreContextTest.cs (hunk 16 above)
Comment (timrisi) [inline tests/monotouch-test/CoreImage/CoreContextTest.cs]: https://developer.apple.com/reference/opengles/eaglcontext iOS and tvOS only

File: tests/monotouch-test/Foundation/UbiquitousKeyValueStoreTest.cs (hunk 18 above)
Comment (timrisi) [inline tests/monotouch-test/Foundation/UbiquitousKeyValueStoreTest.cs]: Great question!  I'll put it back.

Comment (timrisi) [inline tests/monotouch-test/Foundation/UbiquitousKeyValueStoreTest.cs]: Looking at it, might have removed it because the whole test is inside a #if !__WATCHOS__ block, so this is just ignoring the test for other platforms.

Thoughts anyone?

File: tests/monotouch-test/VideoToolbox/VTCompressionSessionTests.cs (hunk 5 above)
Comment (timrisi) [inline tests/monotouch-test/VideoToolbox/VTCompressionSessionTests.cs]: Done: https://bugzilla.xamarin.com/show_bug.cgi?id=51258

File: tests/monotouch-test/Foundation/CalendarTest.cs [hunk 28]
```diff
@@ -18,6 +18,10 @@
 using nuint=global::System.UInt32;
 #endif
 
+#if MONOMAC
```
Comment (chamons) [inline tests/monotouch-test/Foundation/CalendarTest.cs]: Should this be PlatformException and defined on MONOTOUCH and MONOMAC? https://github.com/xamarin/xamarin-macios/blob/master/tools/common/Application.cs#L15

File: tests/monotouch-test/Foundation/FileManagerTest.cs [hunk 29]
```diff
@@ -120,6 +129,9 @@ public void DefaultManager ()
 		}
 
 		[Test]
+#if MONOMAC
```
Comment (chamons) [inline tests/monotouch-test/Foundation/FileManagerTest.cs]: Should we #if the [Test] out instead since this will never be fixed? That way it doesn't show up as ignored?

File: tests/monotouch-test/Foundation/KeyedUnarchiverTest.cs [hunk 30]
```diff
@@ -14,8 +14,13 @@
 #if XAMCORE_2_0
 using Foundation;
 using ObjCRuntime;
+#if MONOMAC
+using AppKit;
+using MonoTouchException=Foundation.ObjCException;
```
Comment (chamons) [inline tests/monotouch-test/Foundation/KeyedUnarchiverTest.cs]: %s/MonoTouchException/PlatformException/gc ?

File: tests/monotouch-test/Foundation/ObjectTest.cs [hunk 31]
```diff
@@ -16,9 +16,15 @@
 using Foundation;
 using CoreGraphics;
 using ObjCRuntime;
-using MonoTouchException=Foundation.MonoTouchException;
 using Security;
+#if MONOMAC
+using AppKit;
+using MonoTouchException = Foundation.ObjCException;
```
Comment (chamons) [inline tests/monotouch-test/Foundation/ObjectTest.cs]: Again on exception.

File: tests/monotouch-test/Foundation/UbiquitousKeyValueStoreTest.cs (hunk 18 above)
Comment (chamons) [inline tests/monotouch-test/Foundation/UbiquitousKeyValueStoreTest.cs]: Maybe remove and do in another PR?

File: tests/monotouch-test/Intents/INIntentResolutionResultTests.cs [hunk 32]
```diff
@@ -37,34 +37,147 @@ public void INIntentResolutionResultIsAbstractTest ()
 		}
 
 		[Test]
-		public void INBooleanResolutionResultPropertyTest ()
```
Comment (chamons) [inline tests/monotouch-test/Intents/INIntentResolutionResultTests.cs]: This is a big nasty diff. Can we reorder it back to the way it was so I can see the diff?

File: tests/monotouch-test/ObjCRuntime/RegistrarTest.cs [hunk 33]
```diff
@@ -5,12 +5,21 @@
...
-#if !__TVOS__ && !__WATCHOS__
+#if !__TVOS__ && !__WATCHOS__ && !MONOMAC
 using AddressBook;
 using AddressBookUI;
 #endif
 using Foundation;
+#if MONOMAC
+using AppKit;
+using UIColor = AppKit.NSColor;
+using MonoTouchException = ObjCRuntime.RuntimeException;
```
Comment (chamons) [inline tests/monotouch-test/ObjCRuntime/RegistrarTest.cs]: Another PlatformException?

File: tests/monotouch-test/Security/KeyTest.cs [hunk 34]
```diff
@@ -88,6 +92,9 @@ public void Encrypt_New ()
 		}
 
 		[Test]
+#if MONOMAC
+		[Ignore ("512 bit encryption fails on mac")]
```
Comment (chamons) [inline tests/monotouch-test/Security/KeyTest.cs]: Bug #?

File: tests/monotouch-test/Security/KeyTest.cs [hunk 35]
```diff
@@ -205,6 +224,9 @@ public void RoundtripRSA1024OAEP ()
...
```
Comment (chamons) [inline tests/monotouch-test/Security/KeyTest.cs]: Bug #?

File: tests/monotouch-test/Intents/INIntentResolutionResultTests.cs (hunk 32 above)
Comment (chamons) [inline tests/monotouch-test/Intents/INIntentResolutionResultTests.cs]: Nevermind. I missed that you were condensing #if blocks.

File: tests/monotouch-test/Foundation/CalendarTest.cs (hunk 28 above)
Comment (timrisi) [inline tests/monotouch-test/Foundation/CalendarTest.cs]: I changed this in the last commit (after your comment), but doesn't seem to be showing that here like it did for tha ll the other comments...

File: tests/monotouch-test/MediaToolbox/FormatNamesTest.cs [hunk 36]
```diff
@@ -25,9 +27,14 @@ public class FormatNamesTest {
 		[Culture ("en")]
 		public void LocalizedNameForMediaType ()
 		{
+#if MONOMAC
+			if (!TestRuntime.CheckMacSystemVersion (10, 11))
+				Assert.Ignore ("Requires macOS 10.11");
+#else
```
Comment (rolfbjarne) [inline tests/monotouch-test/MediaToolbox/FormatNamesTest.cs]: Same: replace `CheckSystemVersion` with call to `TestRuntime.CheckXcodeVersion`.

File: tests/monotouch-test/MediaToolbox/FormatNamesTest.cs [hunk 37]
```diff
@@ -45,9 +52,14 @@ public void LocalizedNameForMediaType ()
 		[Culture ("en")]
 		public void LocalizedNameForMediaSubType ()
 		{
+#if MONOMAC
+			if (!TestRuntime.CheckMacSystemVersion (10, 11))
```
Comment (rolfbjarne) [inline tests/monotouch-test/MediaToolbox/FormatNamesTest.cs]: Same: replace `CheckSystemVersion` with call to `TestRuntime.CheckXcodeVersion`.

File: tests/monotouch-test/ModelIO/MDLLight.cs [hunk 38]
```diff
@@ -41,6 +43,10 @@ public class MDLLightTest {
 		[TestFixtureSetUp]
 		public void Setup ()
 		{
+#if MONOMAC
+			if (!TestRuntime.CheckMacSystemVersion (10, 11))
```
Comment (rolfbjarne) [inline tests/monotouch-test/ModelIO/MDLLight.cs]: Same: replace `CheckSystemVersion` with call to `TestRuntime.CheckXcodeVersion`.

File: tests/monotouch-test/ModelIO/MDLMaterialProperty.cs [hunk 39]
```diff
@@ -39,6 +44,10 @@ public class MDLMaterialPropertyTest {
 		[TestFixtureSetUp]
 		public void Setup ()
 		{
+#if MONOMAC
+			if (!TestRuntime.CheckMacSystemVersion (10, 11))
```
Comment (rolfbjarne) [inline tests/monotouch-test/ModelIO/MDLMaterialProperty.cs]: Same: replace `CheckSystemVersion` with call to `TestRuntime.CheckXcodeVersion`.

File: tests/monotouch-test/ModelIO/MDLTexture.cs [hunk 40]
```diff
@@ -41,6 +40,7 @@ public void Setup ()
 		{
 			TestRuntime.AssertXcodeVersion (7, 0);
 
+#if !MONOMAC
```
Comment (rolfbjarne) [inline tests/monotouch-test/ModelIO/MDLTexture.cs]: The bug this section is about is also present in i386 on mac, so instead of #ifdefing the whole section, you should only ifdef the simulator check.

File: tests/monotouch-test/ObjCRuntime/RegistrarTest.cs [hunk 41]
```diff
@@ -75,6 +84,7 @@ public class RegistrarTest {
 			}
 		}
 
+#if !MONOMAC
```
Comment (rolfbjarne) [inline tests/monotouch-test/ObjCRuntime/RegistrarTest.cs]: Were these tests failing?

File: tests/monotouch-test/SceneKit/GeometrySourceTest.cs [hunk 42]
```diff
@@ -37,11 +41,16 @@ public class GeometrySourceTest {
 		[Test]
 		public void FromMetalBuffer ()
 		{
+#if MONOMAC
+			if (!TestRuntime.CheckMacSystemVersion (10, 11))
```
Comment (rolfbjarne) [inline tests/monotouch-test/SceneKit/GeometrySourceTest.cs]: Same: replace `CheckSystemVersion` with call to `TestRuntime.CheckXcodeVersion`.

File: tests/monotouch-test/CoreBluetooth/UuidTest.cs [hunk 43]
```diff
@@ -35,7 +39,11 @@ public void Roundtrip_16bits ()
 			using (CBUUID uuid = CBUUID.FromString ("1234")) {
 				Assert.That (uuid.Handle, Is.Not.EqualTo (IntPtr.Zero), "Handle");
 				Assert.IsNotNull (uuid.Data, "Data");
-				var expected = UIDevice.CurrentDevice.CheckSystemVersion (7,1) ? "1234" : "Unknown (<1234>)";
+#if MONOMAC
+				var expected = "Unknown (<1234>)";
+#else
+				var expected = UIDevice.CurrentDevice.CheckSystemVersion (7, 1) ? "1234" : "Unknown (<1234>)";
```
Comment (timrisi) [inline tests/monotouch-test/CoreBluetooth/UuidTest.cs]: This one is returning the "Unknown (<1234>)" on macOS 10.12.  Changing the CheckSystemVersion to CheckXcodeVersion would have it looking for "1234"

File: tests/monotouch-test/CoreBluetooth/UuidTest.cs [hunk 44]
```diff
@@ -51,7 +59,11 @@ public void Roundtrip_128bits ()
 			using (CBUUID uuid = CBUUID.FromString ("12345678-90AB-CDEF-cafe-c80c20443d0b")) {
 				Assert.That (uuid.Handle, Is.Not.EqualTo (IntPtr.Zero), "Handle");
 				Assert.IsNotNull (uuid.Data, "Data");
+#if MONOMAC
+				var expected = "Unknown (<12345678 90abcdef cafec80c 20443d0b>)";
```
Comment (timrisi) [inline tests/monotouch-test/CoreBluetooth/UuidTest.cs]: Same issue as above.  MacOS 10.12 is returning the non-xcode 5.1 value.

File: tests/monotouch-test/ObjCRuntime/RegistrarTest.cs (hunk 41 above)
Comment (timrisi) [inline tests/monotouch-test/ObjCRuntime/RegistrarTest.cs]: I don't remember now why I #ifed these out (which is why most places I commented out tests completely I tried to comment why I did. Apparently didn't here). Looking into it now.

Comment (timrisi) [inline tests/monotouch-test/ObjCRuntime/RegistrarTest.cs]: So the reason I #ifed these out is because they crash with a SIGSEGV.  In the Callproperty method these tests use, there's a call to:

`if (Runtime.Arch == Arch.DEVICE && expectFailure) {
  				Console.WriteLine ("Skipping '{0}', it's expected to throw a 'Selector not found exception', but on device it seems to crash instead", selector);
  				return;
  			}`

(which got commented out since mac doesn't have Runtime.Arch), so it looks like it crashses on device on iOS as well.  I could remove the #ifs to let the tests run but fix this call so it just returns instead of skipping completely, but then we just get a bunch of CWLs for the skipped tests and no extra useful data since we don't have a simulator to run it on like iOS does.

Comment (rolfbjarne) [inline tests/monotouch-test/ObjCRuntime/RegistrarTest.cs]: OK

**Comment by timrisi** [comment]
build
